Description: Update for new libav
 .
 vtk (5.10.1-SprezzOS1) unstable; urgency=low
 .
   * Release to unstable
Author: Nick Black <nick.black@sprezzatech.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- vtk-5.10.1.orig/IO/vtkFFMPEGWriter.cxx
+++ vtk-5.10.1/IO/vtkFFMPEGWriter.cxx
@@ -140,7 +140,7 @@ int vtkFFMPEGWriterInternal::Start()
   strcpy(this->avFormatContext->filename, this->Writer->GetFileName());
 
   //create a stream for that file
-  this->avStream = av_new_stream(this->avFormatContext, 0);
+  this->avStream = avformat_new_stream(this->avFormatContext, 0);
   if (!this->avStream) 
     {
     vtkGenericWarningMacro (<< "Could not create video stream.");
@@ -193,7 +193,7 @@ int vtkFFMPEGWriterInternal::Start()
     }
 
   //apply the chosen parameters
-  if (av_set_parameters(this->avFormatContext, NULL) < 0)
+  if (avformat_set_parameters(this->avFormatContext, NULL) < 0)
     {
     vtkGenericWarningMacro (<< "Invalid output format parameters." );
     return 0;
@@ -206,7 +206,7 @@ int vtkFFMPEGWriterInternal::Start()
     vtkGenericWarningMacro (<< "Codec not found." );
     return 0;
     }
-  if (avcodec_open(c, codec) < 0) 
+  if (avcodec_open2(c, codec, NULL) < 0) 
     {
     vtkGenericWarningMacro (<< "Could not open codec.");
     return 0;
@@ -259,14 +259,14 @@ int vtkFFMPEGWriterInternal::Start()
 
 
   //Finally, open the file and start it off.
-  if (url_fopen(&this->avFormatContext->pb, this->avFormatContext->filename, URL_WRONLY) < 0) 
+  if (avio_open(&this->avFormatContext->pb, this->avFormatContext->filename, URL_WRONLY) < 0) 
     {
     vtkGenericWarningMacro (<< "Could not open " << this->Writer->GetFileName() << "." );
     return 0;
     }
   this->openedFile = 1;
 
-  av_write_header(this->avFormatContext);
+  avformat_write_header(this->avFormatContext);
   return 1;
 }
 
@@ -391,7 +391,7 @@ void vtkFFMPEGWriterInternal::End()
 #ifdef VTK_FFMPEG_OLD_URL_FCLOSE
       url_fclose(&this->avFormatContext->pb);
 #else
-      url_fclose(this->avFormatContext->pb);
+      avio_close(this->avFormatContext->pb);
 #endif
       this->openedFile = 0;
       }
